packages/ckeditor5-html-support/src/htmlcomment.ts

```diff
@@ -71,7 +79,7 @@ export default class HtmlComment extends Plugin {
 
 		// Remove comments' markers and their corresponding $root attributes, which are no longer present.
 		editor.model.document.registerPostFixer( writer => {
-			const root = editor.model.document.getRoot()!;
+			const root = editor.model.document.getRoot( editor.model.document.selection.getFirstRange()!.root.rootName! )!;
```

Comment:
This is incorrect. Every HTML-comment-related attribute should be set on the root that contains the comment.

AFAICS, `root` is used only at one place:

```javascript
writer.removeAttribute( marker.name, root );
```

It looks to me that there's nothing wrong with using `marker.getRange().root`.